### PR TITLE
Item resource

### PR DESCRIPTION
**Pull request by nonococoleo**
# Pull Request Template

In this PR, shopcart item has been Refactored to resources, related api path has changed

## Checklist:

- [x] I have written test cases for this change, and my code passed the test
- [x] I have updated documentation to reflect this change
- [x] I have attached this PR to an issue/user story

### CODE REVIEW

File: service/routes.py [hunk 1]
```diff
@@ -316,26 +316,82 @@ def get_shopcart_items(shopcart_id):
...
+        GET /shopcart/{id}/items/{id} - Returns a shopcart Item with the id
+        DELETE /shopcart/{id}/items/{id} -  Deletes a shopcart Item with the id
+        PUT /shopcart/{id}/items/{id} -  Updates a shopcart Item with the id
     """
-    logger.info("Request to get an item in a shopcart")
-    check_content_type("application/json")
 
-    shopcart_item = ShopcartItem.find(item_id)
+    @api.doc('get_shopcart_item')
+    @api.response(404, 'Shopcart Item not found')
```
Comment (iLtc) [inline service/routes.py:332]: Please add a 200 response here

File: service/routes.py [hunk 2]
```diff
@@ -316,26 +316,82 @@ def get_shopcart_items(shopcart_id):
...
+            api.abort(status.HTTP_404_NOT_FOUND,
+                      "Shopcart item with ID [%s] not found in shopcart [%s]." % (item_id, shopcart_id)
+                      )
+
+        logger.info("Fetched shopcart item with ID [%s].", item_id)
+        return shopcart_item.serialize(), status.HTTP_200_OK
+
+    @api.doc('update_shopcart_item')
+    @api.response(404, 'Shopcart Item not found')
+    @api.response(400, 'The posted Item data was not valid')
```
Comment (iLtc) [inline service/routes.py:354]: Please add a 200 response here

File: service/routes.py [hunk 3]
```diff
@@ -316,26 +316,82 @@ def get_shopcart_items(shopcart_id):
...
+        """
+        logger.info("Request to update Shopcart item with id: %s", item_id)
+
+        shopcart_item = ShopcartItem.find(item_id)
+
+        if shopcart_item is None or shopcart_item.sid != shopcart_id:
+            logger.info("Shopcart item with ID [%s] not found in shopcart [%s].", item_id, shopcart_id)
+            api.abort(status.HTTP_404_NOT_FOUND, "Shopcart item with id '{}' was not found.".format(item_id))
+
+        data = request.get_json()
```
Comment (iLtc) [inline service/routes.py:370]: Please use `api.payload` instead of `request.get_json()`

**Review by iLtc — CHANGES_REQUESTED** [review]

**Review by KHafner — APPROVED** [review]

**Review by iLtc — APPROVED** [review]
LGTM, thanks!

**Comment by KHafner** [comment]
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/b5c2987e-b3d1-4601-9f28-34419832137d?env_id=ibm:yp:us-south "nyu-shopcart-service-f20-toolchain"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/c869c127-ceda-423d-80b8-f74ee52b7378 "nyu-shopcart-service-f20-pipeline") deployed [shopcarts](https://github.com/NYU-DevOps2020-shopcarts/shopcarts/releases/tag/deploy-ibm-yp-us-south-kfh226-nyu.edu-dev-20201126-013055 "deploy-ibm-yp-us-south-kfh226-nyu.edu-dev-20201126-013055") to [dev](https://console.bluemix.net/apps/4aa42608-55fd-40d6-b27b-d99aa18a0658?env_id=ibm:yp:us-south "ibm:yp:us-south:kfh226@nyu.edu:dev"), including commit e391d7b2aea41a9554805348c53b2234d191cc5b

**Comment by KHafner** [comment]
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/b5c2987e-b3d1-4601-9f28-34419832137d?env_id=ibm:yp:us-south "nyu-shopcart-service-f20-toolchain"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/c869c127-ceda-423d-80b8-f74ee52b7378 "nyu-shopcart-service-f20-pipeline") deployed [shopcarts](https://github.com/NYU-DevOps2020-shopcarts/shopcarts/releases/tag/deploy-ibm-yp-us-south-kfh226-nyu.edu-prod-20201208-144028 "deploy-ibm-yp-us-south-kfh226-nyu.edu-prod-20201208-144028") to [prod](https://console.bluemix.net/apps/ac06a6b3-040d-43d7-b290-6bfbba3f8c41?env_id=ibm:yp:us-south "ibm:yp:us-south:kfh226@nyu.edu:prod"), including commit e391d7b2aea41a9554805348c53b2234d191cc5b

**Pull request merged**